Network/Converter/PacketConverter.cs

```diff
@@ -27,321 +28,680 @@
-        /// <param name="packet">The packet to convert.</param>
-        /// <returns>System.Byte[].</returns>
-        public byte[] GetBytes(Packet packet)
+        private readonly object packetPropertyCacheLock = new object();
+
+        #endregion Variables
+
+        #region Methods
+
+        /// <summary>
+        /// Returns an array of the <see cref="PropertyInfo"/>s that need to be
+        /// serialised on the given <see cref="Type"/>. If the given <see cref="Type"/>
+        /// has already been cached, it will use the cached <see cref="PropertyInfo"/>
+        /// array, to save CPU time.
+        /// </summary>
+        /// <param name="type">
+        /// The <see cref="Type"/> whose serialisable properties to get.
+        /// </param>
+        /// <returns>
+        /// An array of all <see cref="PropertyInfo"/>s that should be serialised.
+        /// </returns>
+        private PropertyInfo[] GetTypeProperties(Type type)
+        {
+            lock (packetPropertyCacheLock)
+            {
+                // cache the properties to serialise if we haven't already
+                if (!packetPropertyCache.ContainsKey(type))
+                {
+                    packetPropertyCache[type] =
+                        PacketConverterHelper.GetTypeProperties(type);
+                }
+
+                return packetPropertyCache[type];
+            }
+        }
+
+        #region Implementation of IPacketConverter
+
+        /// <inheritdoc />
+        public byte[] SerialisePacket(Packet packet)
+        {
+            MemoryStream memoryStream = new MemoryStream();
+            BinaryWriter binaryWriter = new BinaryWriter(memoryStream);
+
+            SerialiseObjectToWriter(packet, binaryWriter);
+
+            return memoryStream.ToArray();
+        }
+
+        /// <inheritdoc />
+        public byte[] SerialisePacket<P>(P packet) where P : Packet
         {
             MemoryStream memoryStream = new MemoryStream();
             BinaryWriter binaryWriter = new BinaryWriter(memoryStream);
-            GetBytesFromCustomObject(packet, binaryWriter);
+
+            SerialiseObjectToWriter(packet, binaryWriter);
+
             return memoryStream.ToArray();
         }
 
-        /// <summary>
-        /// Applies the byte array to the packets properties.
-        /// </summary>
-        /// <param name="packetType">The type of the final packet.</param>
-        /// <param name="data">The data which should be applied.</param>
-        public Packet GetPacket(Type packetType, byte[] data)
+        /// <inheritdoc />
+        public Packet DeserialisePacket(Type packetType, byte[] serialisedPacket)
         {
-            Packet packet = CreateInstanceOfPacketType(packetType);
-            MemoryStream memoryStream = new MemoryStream(data, 0, data.Length);
-            BinaryReader binaryReader = new BinaryReader(memoryStream);
-            ReadObjectFromStream(packet, binaryReader);
+            MemoryStream memoryStream =
```

Comment:
Already written, but again. No new line after declaration.

Reply:
Was keeping in line with 80 column limit, will change